records.go

```diff
@@ -3,21 +3,30 @@ package dht
 import (
 	"context"
 	"fmt"
+	"strings"
 	"time"
 
+	"github.com/gogo/protobuf/proto"
+	ds "github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/query"
+	u "github.com/ipfs/go-ipfs-util"
+	"github.com/jbenet/goprocess"
+	ci "github.com/libp2p/go-libp2p-core/crypto"
 	"github.com/libp2p/go-libp2p-core/peer"
 	"github.com/libp2p/go-libp2p-core/routing"
-
-	ci "github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-kad-dht/providers"
+	recpb "github.com/libp2p/go-libp2p-record/pb"
 )
 
-// MaxRecordAge specifies the maximum time that any node will hold onto a record
+// maxNonProviderRecordAge specifies the maximum time that any node will hold onto a record
 // from the time its received. This does not apply to any other forms of validity that
 // the record may contain.
 // For example, a record may contain an ipns entry with an EOL saying its valid
 // until the year 2020 (a great time in the future). For that record to stick around
-// it must be rebroadcasted more frequently than once every 'MaxRecordAge'
-const MaxRecordAge = time.Hour * 36
+// it must be rebroadcasted more frequently than once every 'maxNonProviderRecordAge'
+var maxNonProviderRecordAge = time.Hour * 12
```

Comment:
Change based on answer to this [question](https://github.com/libp2p/go-libp2p-kad-dht/issues/323#issuecomment-523049043)